### PR TITLE
Issues: support setting issue template field values with query

### PR DESCRIPTION
**Pull request by techknowlogick**
In Blender this is used to report a bug from inside the application, with some fields automatically filled in.

Code by @brechtvl

### CODE REVIEW

**Comment by brechtvl** [comment]
I implemented this without checking if it works for all field types. Also not sure if there are any security implications related to calling `ctx.FormString(field.ID)`, if crafting a particular field ID in a template could access some information it shouldn't be able to.

**Comment by wxiaoguang** [comment]
1. The direct accessing of `ctx.FormString(field.ID)` will lead to unsolvable conflicts in the future. At least, the keys need prefixes.
2. `camelCase`

File: routers/web/repo/issue.go [hunk 1]
```diff
@@ -760,6 +760,14 @@ func setTemplateIfExists(ctx *context.Context, ctxDataKey string, possibleFiles
 		ctx.Data[ctxDataKey] = template.Content
 
 		if template.Type() == api.IssueTemplateTypeYaml {
+			// Replace field default values by values from query
+			for _, field := range template.Fields {
+				field_value := ctx.FormString(field.ID)
+				if field_value != "" {
+					field.Attributes["value"] = field_value
```
Comment (yardenshoham) [inline routers/web/repo/issue.go]: `snake_case` to `camelCase`
```suggestion
				fieldValue := ctx.FormString(field.ID)
				if fieldValue != "" {
					field.Attributes["value"] = fieldValue
```

Comment (delvh) [inline routers/web/repo/issue.go]: Also, this potentially allows for cross site scripting attacks by inserting something like
`…/issues/new?textField=<script>alert('You've been hacked);</script>` into the URL.
Or do we ensure that it cannot happen due to go templates security aspect?

**Comment by brechtvl** [comment]
> 1. The direct accessing of `ctx.FormString(field.ID)` will lead to unsolvable conflicts in the future. At least, the keys need prefixes.

In GitHub issue templates this works without prefixes, I'm unsure if compatibility or avoiding conflicts would be considered more important.

**Comment by wxiaoguang** [comment]
> > 1. The direct accessing of `ctx.FormString(field.ID)` will lead to unsolvable conflicts in the future. At least, the keys need prefixes.
> 
> In GitHub issue templates this works without prefixes, I'm unsure if compatibility or avoiding conflicts would be considered more important.

The problem is, Gitea's codebase is crowd-contributed, there could be some strange mechanisms. For example, you can access any page with `lang=xxx` to change the language: https://try.gitea.io/test/?lang=de-DE , it means that the `lang` is not suitable to be a form key. And in the future maybe nobody would remember that "the query parameters will be used by a form directly in a page" and add more global keys ..... Indeed, maybe not that serious, just my opinion.

**Comment by zeripath** [comment]
Regarding Lang not being a suitable form key - well I think the user who does that gets to keep both parts when it breaks. (We probably need to switch out that behaviour anyway as it's janky.)

Now, regarding sec implications - FormString can access both URL query and POST form components.

* URL query components seem definitely fine from a sec perspective - although if this is expanded to the API we will need to censor the token etc.
* POST form components - this would allow access to the CSRF token. I think that's all that should be there on a POST to these forms. I wonder though, could a template be crafted in such a way that a user who is already logged in might be tempted to pass in their username and password as an attempt to relogin? Or even a token? Perhaps we need to censor some fields. (I suspect that GitHub does do some censoring here.)

Regarding rendering I think we should be ok. Everything should already be being properly escaped - if it is not then we've already got a problem without this PR. I'm not completely on top of the bit of code that does this though. 

---

So, I think it would be prudent to censor the _csrf and password fields. That should be enough.

**Comment by zeripath** [comment]
As an aside we should might want to consider censoring the fields at the time of authentication and csrf check - then we know that we're safe and that the information cannot leak.

File: routers/web/repo/issue.go (hunk 1 above)
Comment (wolfogre) [reply routers/web/repo/issue.go]: I don't think we should worry about this:

image

image
image
(So many "so what", it's for dramatic effect only, I'm a polite guy.😁)

**Review by wolfogre — APPROVED** [review]

**Comment by techknowlogick** [comment]
FWIW similar functionality is available for non templated issues, and issues aren't being created via query sting (no action) so I think having this functionality is acceptable. Having the namespaced query string options also prevents any collisions.

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
Please update the branch

**Pull request merged**